api/src/main/resources/org/openmrs/module/dtd/config-1.7.dtd

```diff
@@ -0,0 +1,118 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+	Top level configuration element.
+	-->
+	<!ELEMENT module (
+		(id),
+		(name),
+		(version),
+		(package),
+		(author),
+		(description),
+		(activator),
+		(updateURL?),
+		(require_version?),
+		(require_database_version?),
+		(require_modules?),
+		(aware_of_modules?),
+		(mandatory?),
+		(library*),
+		(extension*),
+		(advice*),
+		(privilege*),
+		(globalProperty*),
+		(dwr?),
+		(servlet*),
+		(filter*),
+		(filter-mapping*),
+		(messages*),
+		(mappingFiles?),
+		(packagesWithMappedClasses?),
+		(conditionalResources?)
```

Comment:
Did you forget to add init param?

Reply:
No init-param already exists as it is used in filters - https://github.com/openmrs/openmrs-core/pull/4415/files/257a8045cc69538d7e92b4c4e1fa4378b826c953#diff-b33d1f48bd89c64e920fae246efd0dfd684bf0a8c31917ef68d87ce779eca802R98

Reply:
Can you throw some light in regards to why, in the above, we have all module config elements apart from only init param?

Reply:
init-param was added a few versions ago and from my understanding it can only appear in the config.xml as a child of a `servlet` or a `filter`. It only needs to go in the below module element if it can be a child of module:

ELEMENT module (
    (id),
    (name),
    ...
    (conditionalResources?)
)

If that's wrong and we support init-params as children of the top level module I can add it there too. btw I just pushed a small correction as I missed updating this line to 1.7

	<!ATTLIST module configVersion CDATA #FIXED "1.7">

Reply:
You are absolutely correct! Thanks for the clarification! 👍